src/components/hoc-helpers/withHeaderName.js

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+
+const withHeaderName = (View) => {
+  return class extends Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+          controlHeader: '',
+          setControlHeader: this.setControlHeader
```

Comment:
why do you save a method on the state?

Reply:
to be able to pass it as props

Reply:
see below how to pass it without putting it in state :)

Reply:
I removed the HOC as it is redundant. Done